dashboard/modules/metrics/metrics_head.py

```diff
@@ -135,8 +140,23 @@ async def get_progress(self, req):
         then we will fetch the progress across all jobs.
         """
         job_id = req.query.get("job_id")
+
+        job = await find_job_by_ids(
+            self._dashboard_head.gcs_aio_client, self._job_manager, job_id
+        )
         job_id_query = f'{{JobId="{job_id}"}}' if job_id else ""
-        query = f"sum(ray_tasks{job_id_query}) by (State)"
+
+        if job and job.status in [
+            JobStatus.STOPPED,
+            JobStatus.SUCCEEDED,
+            JobStatus.FAILED,
+        ]:
+            # Because ray does not keep metrics after a job dies,
+            # we fetch the max_over_time over last 14 days
+            query = f"sum(max_over_time(ray_tasks{job_id_query}[14d])) by (State)"
+        else:
+            query = f"sum(ray_tasks{job_id_query}) by (State)"
```

Comment:
I think you actually need to perform both fetches all the time, since FAILED/SUCCEEDED time series could be erased when workers exit. So the queries should be `sum(max_over_time(ray_tasks{job_id_query,State~="FINISHED|FAILED}[14d])) by (State)` and `sum(ray_tasks{job_id_query,State!~"FINISHED|FAILED"}) by (State)`.